### DISCUSSION TITLE
Using access tokens on supercomputers (distributed machine)

### DISCUSSION THREAD

**Discussion by Weiming-Hu** [community/community#21605] [New to GitHub]
<p>I have recently signed up for two-factor authentication. I’m aware that, on my personal computer, I can set up access token to make it “remember” the password.</p>
<p>How would I do this on a distributed machine with several servers as login nodes? I have tried to generate access tokens every time I use git commands. But these servers have different hostname and I’m not sure whether I land on the same machine next time I log onto the supercomputer.</p>
<p>Does anyone have any pointers? Thank you.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>If you’re talking about personal access tokens, those are not bound to a specific machine. You could store it in a password manager on your local machine and enter it from there into your session on the supercomputer (assuming you’re using SSH or similar to log in). You could <a href="https://docs.github.com/en/github/getting-started-with-github/getting-started-with-git/caching-your-github-credentials-in-git">use a credential cache</a> on the supercomputer so you don’t have to enter the token for each and every Git operation.</p>

**Comment by Weiming-Hu** [comment]
<p>I see. I actually find another difference when I did things locally and on supercomputers. If I use the SSH based git URL, I can avoid entering keyword every time as long as I have added the public key to my GitHub account. Thank you. I have stored the personal access token locally so that I can use it for the HTTPS based repo that I have already cloned.</p>
